packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3121,17 +3138,29 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         ackHandle: string,
         summaryRefSeq: number,
         summaryLogger: ITelemetryLogger,
+        fetchSource?: FetchSource,
     ) {
+        // If we don't want to use cache and fetch latest from network, then just use the update from server flow.
+        if (fetchSource === FetchSource.noCache) {
+            await this.refreshLatestSummaryAckFromServer(summaryLogger);
+            return;
+        }
         const readAndParseBlob = async <T>(id: string) => readAndParse<T>(this.storage, id);
```

Comment:
When do we need the code below?
Shouldn't we always do this.refreshLatestSummaryAckFromServer() ?
Fetching particular version is more expensive than fetching latest one.
That said, are we ignoring proposalHandle / proposalHandle when fetchSource === FetchSource.noCache? It looks weird - it feels like either API shape (arguments) should be changed to allow one ask or the other, or we should break it into two APIs, but as it stands (after changes), the argument shape does not make sense

Reply:
I think we should just remove this function and instead just use refreshLatestSummaryAckFromServer to always refresh with latest. See my other comment.